### PR TITLE
minio: 20170613 -> 2017-09-29T19-16-56Z

### PR DESCRIPTION
**Pull request by bachp**
###### Motivation for this change

Update to lastest version: [CHANGELOG](https://github.com/minio/minio/releases/tag/RELEASE.2017-08-05T00-00-53Z)

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/servers/minio/default.nix [hunk 1]
```diff
@@ -3,12 +3,12 @@
 stdenv.mkDerivation rec {
   name = "minio-${shortVersion}";
 
-  shortVersion = "20170613";
-  longVersion = "2017-06-13T19-01-01Z";
+  shortVersion = "20170805";
+  longVersion = "2017-08-05T00-00-53Z";
```
Comment (orivej) [inline pkgs/servers/minio/default.nix]: Could you delete `shortVersion` and rename `longVersion` to `version`?

File: pkgs/servers/minio/default.nix [hunk 2]
```diff
@@ -3,12 +3,12 @@
...
 
   src = fetchurl {
     url = "https://github.com/minio/minio/archive/RELEASE.${lib.replaceStrings [":"] ["-"] longVersion}.tar.gz";
```
Comment (orivej) [inline pkgs/servers/minio/default.nix]: `replaceStrings` is not needed, just use `${version}`.

Comment (bachp) [reply pkgs/servers/minio/default.nix]: Done

File: pkgs/servers/minio/default.nix (hunk 1 above)
Comment (bachp) [reply pkgs/servers/minio/default.nix]: The new version now only uses the long version which is now called `version`

File: pkgs/servers/minio/default.nix [hunk 3]
```diff
@@ -1,14 +1,13 @@
 { lib, stdenv, fetchurl, go }:
 
 stdenv.mkDerivation rec {
-  name = "minio-${shortVersion}";
+  name = "minio-${version}";
 
-  shortVersion = "20170613";
-  longVersion = "2017-06-13T19-01-01Z";
+  version = "2017-09-29T19-16-56Z";
```
Comment (Mic92) [inline pkgs/servers/minio/default.nix]: What a weird version scheme. Only milliseconds are missing.

Comment (bachp) [reply pkgs/servers/minio/default.nix]: I guess it allows for a pretty rapid release cycle 😉

**Pull request merged**